src/Loggers/LogImage.jl

```diff
@@ -1,5 +1,20 @@
+# Note: we use the 100s place of the enum to represent the number of dimensions
+# And the 10s place to represent which dimension is the observation (N), 0 for no observation dimension
+@enum ImageFormat begin
+    L=100;
+    CL=200; LC; HW; WH; NL=210; LN=220;
+    HWC=300; WHC; CHW; CWH; NCL=310; NHW; NWH; NLC; CLN=330; LCN; HWN; WHN;
+    NHWC=410; NWHC; NCHW; NCWH; HWCN=440; WHCN; CHWN; CWHN;
+end
+
+expected_ndims(x::ImageFormat) = Int(x) ÷ 100 # get the 100's digit
+obs_dim(x::ImageFormat) = (Int(x) % 100) ÷ 10 # get the 10's digit
+strip_obs = Dict(
```

Comment:
this should be `const`.

Also idk if it is worth doing this, but this is a good use-case for a OrderedCollections.jl `LittleDict`
would would be `const strip_obs = freeze(LittleDict(`

It could also be done with a function of course
```
function strip_obs(format)
	if format in (NL, LN)
		L
	elseif format in (NCL, CLN, NLC, LCN)
		LC
	else if ...
	else 
		format
	end
end
```

Another nice option might be MLStyle.jl
https://thautwarm.github.io/MLStyle.jl/latest/syntax/pattern/#Or-Pattern-1

```
function strip_obs(format)
	@match format begin
	NL || LN => L
	NCL || CLN || NLC || LCN => LC
	...
	_ => format	
	end
end
```

Reply:
`const strip_obs` and `function strip_obs` both show very similar performance. using `@time`. Other options are appealing but then they'll add to dependencies. I'll stick with `const strip_obs` dictionary here.